### PR TITLE
[ENG-4959] Classify notable domains as unverified

### PR DESCRIPTION
**Pull request by Johnetordoff**
## Purpose

Create a new varity of NotableDomain Note for timedout domains.

## Changes

- changes for domain extractor task
- changes to tests.

## QA Notes

Please make verification statements inspired by your code and what your code touches.
- Verify
- Verify

What are the areas of risk?

Any concerns/considerations/questions that development raised?

## Ticket

https://openscience.atlassian.net/browse/ENG-4959

### CODE REVIEW

File: osf_tests/test_notable_domains.py [hunk 1]
```diff
@@ -75,7 +78,7 @@ def test_actract_domains__returned_on_error(self):
         with mock.patch.object(spam_tasks.requests, 'head') as mock_head:
             mock_head.side_effect = spam_tasks.requests.exceptions.Timeout
             domains = set(spam_tasks._extract_domains(sample_text))
-        assert domains == {sample_text}
```
Comment (Johnetordoff) [inline osf_tests/test_notable_domains.py:78]: This is the changed test case, others are just changed to pass the note.

File: osf/models/notable_domain.py [hunk 2]
```diff
@@ -14,6 +14,7 @@ class Note(IntEnum):
         ASSUME_HAM_UNTIL_REPORTED = 1
         UNKNOWN = 2
         IGNORED = 3
+        UNVERIFIED = 4  # Timedout couldn't determine
```
Comment (mfraezz) [inline osf/models/notable_domain.py:17]: Did you run `python3 manage.py makemigrations`? I would expect a schema change in `note.choices` to generate an automigration, even if it's a functional no-op for the database.

**Review by mfraezz — CHANGES_REQUESTED** [review]
A couple small questions/concerns, but once those are resolved this should be RTM :octocat:

File: osf/external/spam/tasks.py [hunk 3]
```diff
@@ -46,8 +46,11 @@ def _check_resource_for_domains(guid, content):
     resource = guid.referent
     spammy_domains = []
     referrer_content_type = ContentType.objects.get_for_model(resource)
-    for domain in _extract_domains(content):
-        notable_domain, _ = NotableDomain.objects.get_or_create(domain=domain)
+    for domain, note in _extract_domains(content):
+        notable_domain, _ = NotableDomain.objects.get_or_create(
+            domain=domain,
+            defaults={'note': note}
+        )
```
[the remark this answers is not in the document]
Comment (Johnetordoff) [reply osf/external/spam/tasks.py:53]: @jwalz can you confirm the `is_triaged` behavior here, I'm not sure the admin workflow ramifications.

[the remark this answers is not in the document]
Comment (jwalz) [reply osf/external/spam/tasks.py:53]: I legit cannot figure out what `is_triaged` is for, but Matt's suggestion seems right

**Comment by mfraezz** [comment]
Squashed into af35c28a493c29161424fc4f9c7df9d662c0e3c1, consider this merged

**Pull request closed without merging**